src/missed_spans.rs

```diff
@@ -57,43 +54,105 @@ impl<'a> FmtVisitor<'a> {
 
         self.last_pos = end;
         let span = codemap::mk_sp(start, end);
-        let snippet = self.snippet(span);
 
-        self.write_snippet(&snippet, true, &process_last_snippet);
+        self.write_snippet(span, true, &process_last_snippet);
     }
 
     fn write_snippet<F: Fn(&mut FmtVisitor, &str, &str)>(&mut self,
-                                                         snippet: &str,
+                                                         span: Span,
                                                          last_snippet: bool,
                                                          process_last_snippet: F) {
+        // FIXME: this is all 1 big turbo mess -- fix asap
+        let big_span = codemap::mk_sp(self.codemap.lookup_char_pos(span.lo).file.start_pos,
+                                      span.hi);
+        let big_snippet = self.snippet(big_span);
+        let big_diff = (span.lo - big_span.lo).0 as usize;
+
+        let snippet = self.snippet(span);
+
```

Comment:
Ah yes, this is a particularly bad hack. Basically we need the chars preceding the current span to check if the comment follows code or a mostly empty line. We trace back to the start of the file, but only look as far as the start of the current line. Because a `&str` is a `DoubleEndedIterator`, this should be fairly efficient (though mega hacky).

I'd love to do this in a more sane way.